custom_components/zwave_mqtt/__init__.py

```diff
@@ -212,6 +215,14 @@ def async_value_removed(value):
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Unload a config entry."""
+    # cleanup device registry (which will in turn clean entity registry)
+    dev_registry = await get_dev_reg(hass)
+    devices_to_remove = [
+        item.id for item in devices_for_config_entry(dev_registry, entry.unique_id)
+    ]
+    for dev_id in devices_to_remove:
+        dev_registry.async_remove_device(dev_id)
```

Comment:
We shouldn't do this. The core is responsible for cleaning the device registry on config entry removal.

Reply:
That's what I was discussing on discord. Accidentally left the code in ;-)
Anyways, with this present, devices are cleaned up, without it, they won't get cleaned out.

Reply:
Let's remove it and look for the bug in the core.

Reply:
fixed